okio/native/src/main/kotlin/okio/ByteString.kt

```diff
@@ -0,0 +1,214 @@
+  internal actual open fun getSize() = commonGetSize()
+
+  /** Returns a byte array containing a copy of the bytes in this `ByteString`. */
+  actual open fun toByteArray() = commonToByteArray()
+
+  /** Returns the bytes of this string without a defensive copy. Do not mutate!  */
+  internal actual open fun internalArray() = commonInternalArray()
+
+  /**
+   * Returns true if the bytes of this in `[offset..offset+byteCount)` equal the bytes of `other` in
+   * `[otherOffset..otherOffset+byteCount)`. Returns false if either range is out of bounds.
+   */
+  actual open fun rangeEquals(
+    offset: Int,
+    other: ByteString,
+    otherOffset: Int,
+    byteCount: Int
+  ): Boolean = commonRangeEquals(offset, other, otherOffset, byteCount)
+
+  /**
+   * Returns true if the bytes of this in `[offset..offset+byteCount)` equal the bytes of `other` in
+   * `[otherOffset..otherOffset+byteCount)`. Returns false if either range is out of bounds.
+   */
+  actual open fun rangeEquals(
+    offset: Int,
+    other: ByteArray,
+    otherOffset: Int,
+    byteCount: Int
+  ): Boolean = commonRangeEquals(offset, other, otherOffset, byteCount)
+
+  actual fun startsWith(prefix: ByteString) = commonStartsWith(prefix)
+
+  actual fun startsWith(prefix: ByteArray) = commonStartsWith(prefix)
+
+  actual fun endsWith(suffix: ByteString) = commonEndsWith(suffix)
+
+  actual fun endsWith(suffix: ByteArray) = commonEndsWith(suffix)
+
+  actual fun indexOf(other: ByteString, fromIndex: Int) = indexOf(other.internalArray(), fromIndex)
+
+  actual open fun indexOf(other: ByteArray, fromIndex: Int) = commonIndexOf(other, fromIndex)
+
+  // TODO move lastIndexOf() when https://youtrack.jetbrains.com/issue/KT-22818 is fixed
+
+  fun lastIndexOf(other: ByteString, fromIndex: Int = size) = lastIndexOf(other.internalArray(),
+    fromIndex)
+
+  open fun lastIndexOf(other: ByteArray, fromIndex: Int = size): Int {
+    var fromIndex = fromIndex
+    fromIndex = minOf(fromIndex, data.size - other.size)
+    for (i in fromIndex downTo 0) {
+      if (arrayRangeEquals(data, i, other, 0, other.size)) {
+        return i
+      }
+    }
+    return -1
+  }
+
+  actual override fun equals(other: Any?) = commonEquals(other)
+
+  actual override fun hashCode() = commonHashCode()
+
+  actual override fun compareTo(other: ByteString) = commonCompareTo(other)
+
+  /**
+   * Returns a human-readable string that describes the contents of this byte string. Typically this
+   * is a string like `[text=Hello]` or `[hex=0000ffff]`.
+   */
+  actual override fun toString(): String = data.toUtf8String()
+
+  actual companion object {
+    internal actual val HEX_DIGITS = COMMON_HEX_DIGITS
+
+    /** A singleton empty `ByteString`.  */
+    actual val EMPTY: ByteString = COMMON_EMPTY
+
+    /** Returns a new byte string containing a clone of the bytes of `data`. */
+    actual fun of(vararg data: Byte) = commonOf(*data.copyOf())
```

Comment:
Build conflict with #471 if both merged. If you merge one, I'll update the other.